### PR TITLE
UX-1070/storage listing view created

### PR DESCRIPTION
**Pull request by rajshekhar23**
image

image

image

image

### CODE REVIEW

File: src/api-client/Qbert.ts [hunk 1]
```diff
@@ -343,6 +347,57 @@ class Qbert extends ApiService {
...
+    return data.items && data.items.map(this.convertResource(clusterId))
+  }
+
+  @trackApiMethodMetadata({
+    url: '/clusters/:clusterId/k8sapi/api/v1/namespaces/default/persistentvolumeclaims',
+    type: 'GET',
+    params: ['clusterId'],
+  })
+  getPersistentVolumeClaims = async (clusterId) => {
+    const url = `/clusters/${clusterId}/k8sapi/api/v1/namespaces/default/persistentvolumeclaims`
```
Comment (jruddell) [inline src/api-client/Qbert.ts:390]: Do PVC's need a route filtered by namespace or is there a cluster level api? The namespace shouldn't be hardcoded as well. Look at how we pass namespaces to other apis

Comment (rajshekhar23) [reply src/api-client/Qbert.ts:390]: ok. Will update.

File: src/app/plugins/kubernetes/components/storage/selectors.ts [hunk 2]
```diff
@@ -25,3 +25,85 @@ export const storageClassSelector = createSelector(
...
+      'clusterId',
+    ]),
+    clustersSelector,
+    importedClustersSelector,
+  ],
+  (rawPersistentVolumeClaims, clusters, importedClusters) => {
+    console.log(rawPersistentVolumeClaims)
+    return rawPersistentVolumeClaims
+      .map((persistentVolumeClaim) => ({
+        id: persistentVolumeClaim?.metadata?.uid,
```
Comment (knnguy) [inline src/app/plugins/kubernetes/components/storage/selectors.ts:91]: Remove id, name, and namespace here. We already surfaced those fields in the `convertResource` function in Qbert

File: src/app/plugins/kubernetes/components/storage/selectors.ts [hunk 3]
```diff
@@ -25,3 +25,85 @@ export const storageClassSelector = createSelector(
...
+        name: persistentVolumeClaim?.metadata?.name,
+        namespace: persistentVolumeClaim?.metadata?.namespace,
+        labels: Object.values(persistentVolumeClaim?.metadata?.labels).join(','),
```
Comment (knnguy) [inline src/app/plugins/kubernetes/components/storage/selectors.ts:94]: We should probably do the join (' , ') in view

**Review by knnguy — DISMISSED** [review]

File: src/api-client/Qbert.ts (hunk 1 above)
Comment (knnguy) [reply src/api-client/Qbert.ts:390]: There is. The request url is `const url = `/clusters/${clusterId}/k8sapi/api/v1/namespaces``

File: src/app/plugins/kubernetes/components/storage/persistentVolumeClaims/actions.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import createCRUDActions from 'core/helpers/createCRUDActions'
+import { parseClusterParams } from 'k8s/components/infrastructure/clusters/actions'
+import { ActionDataKeys } from 'k8s/DataKeys'
+import { flatten, pluck } from 'ramda'
+import { someAsync } from 'utils/async'
+import { persistentVolumeClaimsSelector } from '../selectors'
+
+const { qbert } = ApiClient.getInstance()
+
+const persistentVolumeClaimsActions = createCRUDActions(ActionDataKeys.PersistentVolumeClaims, {
```
[the remark this answers is not in the document]
Comment (rajshekhar23) [reply src/app/plugins/kubernetes/components/storage/persistentVolumeClaims/actions.ts:13]: Updated

File: src/app/plugins/kubernetes/components/storage/persistentVolumeClaims/actions.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    if (clusterId === allKey) {
+      return someAsync(
+        pluck<any, string>('uuid', clusters).map(qbert.getPersistentVolumeClaims),
+      ).then(flatten)
+    }
+    return qbert.getPersistentVolumeClaims(clusterId)
+  },
+  uniqueIdentifier: 'metadata.uid',
+  indexBy: 'clusterId',
+  entityName: 'CSI Drivers',
```
[the remark this answers is not in the document]
Comment (rajshekhar23) [reply src/app/plugins/kubernetes/components/storage/persistentVolumeClaims/actions.ts:26]: Updated  entityName as 'CSIDrivers'

File: src/app/plugins/kubernetes/components/storage/selectors.ts (hunk 2 above)
Comment (rajshekhar23) [reply src/app/plugins/kubernetes/components/storage/selectors.ts:91]: Removed.

File: src/app/plugins/kubernetes/components/storage/selectors.ts (hunk 3 above)
Comment (rajshekhar23) [reply src/app/plugins/kubernetes/components/storage/selectors.ts:94]: Updated

**Review by jruddell — APPROVED** [review]

**Pull request merged**